src/lib/logical_query_plan/join_node.cpp

```diff
@@ -71,6 +71,54 @@ std::vector<std::shared_ptr<AbstractExpression>> JoinNode::output_expressions()
   return output_expressions;
 }
 
+std::shared_ptr<LQPUniqueConstraints> JoinNode::unique_constraints() const {
+  // Semi- and Anti-Joins act as mere filters for input_left().
+  // Therefore, existing constraints are forwarded as they remain valid.
+  if (join_mode == JoinMode::Semi || join_mode == JoinMode::AntiNullAsTrue || join_mode == JoinMode::AntiNullAsFalse) {
+    return _forward_left_unique_constraints();
+  }
+
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+
+  // No guarantees for Cross Joins and multi predicate joins
+  const auto predicates = join_predicates();
+  if (predicates.empty() || predicates.size() > 1) return unique_constraints;
+  // Also, no guarantees for other join predicates than _equals() (Equi Join)
+  const auto join_predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(join_predicates().front());
+  if (!join_predicate || join_predicate->predicate_condition != PredicateCondition::Equals) return unique_constraints;
+
+  DebugAssert(join_mode == JoinMode::Inner || join_mode == JoinMode::Left || join_mode == JoinMode::Right ||
+                  join_mode == JoinMode::FullOuter,
+              "Unhandled JoinMode");
```

Comment:
Mmh, this now looks as if we did not yet implement all join modes. I'd move it below the if block in the beginning.